src/core/constraints/ShapeBasedConstraint.cpp

```diff
@@ -77,7 +78,7 @@ ParticleForce ShapeBasedConstraint::force(Particle const &p,
         force1 +=
             dpd_pair_force(p, part_rep, ia_params, dist_vec, dist, dist * dist);
         // Additional use of DPD here requires counter increase
-        dpd_rng_counter_increment();
+        dpd.rng_increment();
```

Comment:
shouldn't we increment the counter where the random number is used? actually, we can increment the counter automatically in the function that provides the random number.

Reply:
Don't we want to use the same counter state for all calls to RNGs within the same integration step? Otherwise the trajectory depends on the order particles are traversed.

On a side-note, I think this function actually belongs to `forces.cpp` or similar. The constraint shouldn't need to know anything about thermostats and non-bonded interactions, and should simply pass a reference to its particle representation.